### PR TITLE
Fixed model save and load

### PR DESCRIPTION
**Pull request by shantanuchhabra**

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py [hunk 1]
```diff
@@ -95,12 +95,12 @@ def _get_native_state(self):
...
+        state['detector'] = state['detector']._get_native_state()
         return state
 
     @classmethod
     def _load_version(cls, state, version):
-        assert(version == _PYTHON_ONE_SHOT_OBJECT_DETECTOR_VERSION)
+        assert(version == cls._PYTHON_ONE_SHOT_OBJECT_DETECTOR_VERSION)
         # we need to undo what we did at save and turn the proxy object
         # back into a Python class
         state['detector'] = _ObjectDetector(state['detector'])
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: Have you tested this? Doesn't this need to decode the native state back into an OD model first?

Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: Yeah, I tested `get_augmented_images` and checked for the `detector` key in it and it seemed to work!

Comment (nickjong) [reply src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: It doesn't suffice for the `detector` key to exist. The value must be a (functioning) object detector.

Comment (nickjong) [inline src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: If `state['detector']` is the result of `_get_native_state()`, how does `_ObjectDetector` correctly parse that?

Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: Fixed.

Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py]: Fixed!

**Pull request merged**